plugin.rb

```diff
@@ -502,9 +503,10 @@ class ::Guardian
     def self.reset_accepted_answer_cache
       @@allowed_accepted_cache["allowed"] = begin
         Set.new(
-          CategoryCustomField.where(name: "enable_accepted_answers", value: "true").pluck(
-            :category_id,
-          ),
+          CategoryCustomField.where(
+            name: ::DiscourseSolved::ENABLE_ACCEPTED_ANSWERS_CUSTOM_FIELD,
+            value: "true",
```

Comment:
Hmmm... I haven't used it for a long time but isn't it supposed to be queried  with `"t"` or `true` ?


```ruby
[12] pry(main)> CategoryCustomField.create!(category_id: 1, name: "foo", value: true)
=> #<CategoryCustomField:0x0000000106970f20
 id: 7,
 category_id: 1,
 name: "foo",
 value: "t",
 created_at: Fri, 16 Jun 2023 12:32:51.566383000 UTC +00:00,
 updated_at: Fri, 16 Jun 2023 12:32:51.566383000 UTC +00:00>
 [13] pry(main)> CategoryCustomField.where(value: "true").count
  CategoryCustomField Count (1.3ms)  SELECT COUNT(*) FROM "category_custom_fields" WHERE "category_custom_fields"."value" = 'true'
=> 0
[14] pry(main)> CategoryCustomField.where(value: "t").count
  CategoryCustomField Count (0.4ms)  SELECT COUNT(*) FROM "category_custom_fields" WHERE "category_custom_fields"."value" = 't'
=> 1
[15] pry(main)> CategoryCustomField.where(value: true).count
  CategoryCustomField Count (1.2ms)  SELECT COUNT(*) FROM "category_custom_fields" WHERE "category_custom_fields"."value" = 't'
=> 1
```

Reply:
Yeah I had that same thought but I did some test locally and it seems some are "t" and some are "true" actually there is an example here:

```ruby def self.reset_accepted_answer_cache
      @@allowed_accepted_cache["allowed"] = begin
        Set.new(
          CategoryCustomField.where(
            name: ::DiscourseSolved::ENABLE_ACCEPTED_ANSWERS_CUSTOM_FIELD,
            value: "true",
          ).pluck(:category_id),
        )
      end
    end
```

Reply:
Well yes if you do "true":

```
[1] pry(main)> CategoryCustomField.create!(category_id: 1, name: "foo", value: "true")
  TRANSACTION (0.1ms)  BEGIN
  CategoryCustomField Create (1.6ms)  INSERT INTO "category_custom_fields" ("category_id", "name", "value", "created_at", "updated_at") VALUES (1, 'foo', 'true', '2023-06-16 13:37:01.261804', '2023-06-16 13:37:01.261804') RETURNING "id"
  TRANSACTION (0.5ms)  COMMIT
=> #<CategoryCustomField:0x0000000109379a10
 id: 8,
 category_id: 1,
 name: "foo",
 value: "true",
 created_at: Fri, 16 Jun 2023 13:37:01.261804000 UTC +00:00,
 updated_at: Fri, 16 Jun 2023 13:37:01.261804000 UTC +00:00>
 ```
 
 It saves the string, but ultimately here you want a boolean no?

Reply:
If I remmeber correctly the schema for the table has the value field as text but yeah all in all we want a boolean

Reply:
I mean yes you are probably right and this is how the code was written before, so probably ok :+1: